### PR TITLE
RENO-2914: React DS/Reservoir V1 Upgrade WIP

### PR DESCRIPTION
**Pull request by nypl-wluisi**
[Jira Ticket](https://jira.nypl.org/browse/RENO-2914)

**This PR does the following:**
- Updates app from DS v0.25.9 -> v0.28.0
- Updates app for DS string enums change, updating all components that used DS exported enums to use "string literal unions"
- Refactors some components to use typescript to make DS code change easier and more detectable (i.e, tsx files will crash a build and tell you if a prop type is wrong, regular .js files wont)

### Review Steps:
- [ ] Pull in `RENO-2914/ds-v1-upgrade` branch
- [ ] Run `npm install`
- [ ] Start app in production mode: `npm run build && npm start`
- [ ] Manually test different pages in the app, i.e, Blogs, Location Finder, Articles & Databases (aka Online Resources), Locations Request Visit
- [ ] Review code changes via github

### CODE REVIEW

File: src/components/blogs/BlogCollection/BlogCollection.tsx [hunk 1]
```diff
@@ -78,7 +78,6 @@ function BlogCollection({
   slug = "",
   slugLabel = "",
   limit,
-  pageNumber,
```
Comment (isastettler) [inline src/components/blogs/BlogCollection/BlogCollection.tsx:81]: Why do we keep pageNumber in BlogCollectionProps? (line 68)

File: src/components/ds-prototypes/FilterBar/FilterBar.tsx [hunk 2]
```diff
@@ -131,8 +118,8 @@ function FilterBar({
           <Heading
             id="search-filters--heading"
             //className={s.filterBarHeading}
```
Comment (isastettler) [inline src/components/ds-prototypes/FilterBar/FilterBar.tsx:120]: Would love to know more about the replacement of the css.modules. I see that this file still uses some and I assume it has to do with the old design still being used?

**Review by isastettler — APPROVED** [review]
great job!

**Comment by nypl-wluisi** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr273-fhm0gytdnk48fkvwu0xv06gzblouup3s.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/626c3e758a3b7b49ded68328

**Pull request merged**